nixos/modules/services/networking/infiniband.nix

```diff
@@ -0,0 +1,78 @@
+{lib, pkgs, config, ...}:
+
+let
+  cfg = config.networking.ib-interfaces;
+in {
+
+  imports = [];
+
+  options = {
+    networking.ib-interfaces = {
+      enable = lib.mkEnableOption "ib-interfaces";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+
+    environment.systemPackages = [ pkgs.rdma-core ];
+    services.udev.packages =
+      let patchedUdevRules = pkgs.stdenv.mkDerivation {
```

Comment:
It seems like this would be better expressed as `runCommand` as well:

```
runCommand "patched-${pkgs.rdma-core.name}-udev-rules" {
 preferLocalBuild = true;
} ''
  mkdir -p $out/lib/udev
  cp -r {${pkgs.rdma-core},$out}/lib/udev/rules.d
  substituteInPlace $out/lib/udev/rules.d60-srp_daemon.rules \
            --replace /bin/systemctl ${pkgs.systemd}/bin/systemctl
''

Reply:
strangely this change causes my deploy script to hang, so I reverted to the mkDerivation call

Reply:
Is there any reason not to patch these udev rules inside `pkgs/os-specific/linux/rdma-core/default.nix` itself?
Then you could just set services.udev.packages = [ pkgs.rdma-core ]; and be done.

Reply:
Also, I couldn't find this file in the current `rdma-core-29.0` - any idea where it went?

Reply:
@flokli We should patch udev rules directly in `rdma-core`